### DISCUSSION TITLE
TypeScript errors in custom overlay example

### DISCUSSION THREAD

**Discussion by phibo23** [uber/react-map-gl#2274] [Q&A]
I am implementing a custom map control based on the "custom overlay" example: https://github.com/visgl/react-map-gl/blob/f903e44593e67a320fbe3257f0a2b687e7061cbd/examples/custom-overlay/src/custom-overlay.tsx

It works fine and I was able to adapt it to my requirements, but when I build my code (as part of a Next.js project) I am getting some TypeScript errors such as:
<img width="459" alt="Bildschirmfoto 2023-09-07 um 14 35 03" src="https://github.com/visgl/react-map-gl/assets/1328782/3db87e81-2858-41b4-8296-760cdcb3a692">

I am not familiar enough with the types used by the library internally to find a proper solution. Currently I just added `@ts-ignore` to all affected lines.

**Comment by Pessimistress** [comment]
If your tsconfig has strictNullCheck then just type `_map` as `MapInstance | null`.
